docs/_docs/reference/contextual/derivation.md

```diff
@@ -26,17 +26,125 @@ given [T: Show]     : Show[Tree[T]]     = Show.derived
 
 We say that `Tree` is the _deriving type_ and that the `Eq`, `Ordering` and `Show` instances are _derived instances_.
 
-**Note:** The access to `derived` above is a normal access, therefore if there are multiple definitions of `derived` in the type class, overloading resolution applies.
-
-**Note:** `derived` can be used manually, this is useful when you do not have control over the definition. For example we can implement an `Ordering` for `Option`s like so:
+**Note:** `derived` can be used manually, this is useful when you do not have control over the definition. For example we can implement `Ordering` for `Option`s like so:
 
 ```scala
 given [T: Ordering]: Ordering[Option[T]] = Ordering.derived
 ```
 
 It is discouraged to directly refer to the `derived` member if you can use a `derives` clause instead.
 
-All data types can have a `derives` clause. This document focuses primarily on data types which also have a given instance
+## Exact mechanism
+In the following, when type arguments are enumerated and the first index evaluates to a larger value than the last, then there are actually no arguments, for example: `A[T_2, ..., T_1]` means `A`.
+
+For a class/trait/object/enum `DerivingType[T_1, ..., T_N] derives TC`, a derived instance is created in `DerivingType`'s companion object (or `DerivingType` itself if it is an object).
+
+The general "shape" of the derived instance is as follows:
+```scala
+given [...]: TC[ ... DerivingType[...] ... ] = TC.derived
```

Comment:
debatable but its probably better to be explicit then imply hidden context bounds
```suggestion
given [...](using ...): TC[ ... DerivingType[...] ... ] = TC.derived
```

Reply:
I see what you mean, but I think it would be more confusing than anything

Maybe I should add a note in the part where the context bounds are used ?

Reply:
What about the following ?
Adding the using clause to the "general shape", and 
Adding a note to the section with the context bounds reminding they do indeed introduce a using

Reply:
or you can reference the expansion of context bounds from here https://docs.scala-lang.org/scala3/reference/contextual/context-bounds.html

Reply:
I just did this, literally this second, what a coincidence

I'll let you look and tell me what you think

Reply:
yeah it looks good!